logger.go

```diff
@@ -44,6 +45,13 @@ func (l LevelLogger) DebugPrintf(format string, a ...interface{}) {
 	}
 }
 
+func (l LevelLogger) TracePrintf(format string, a ...interface{}) {
+	if l.debug {
+		l.Print("[TRACE] ")
+		l.Printf(format, a...)
```

Comment:
These 2 lines can be merged into

```go
l.Print("[TRACE] " + format, a...)
```

Reply:
Didn't go through the code careful enough .... Actually if you write in two line and there are concurrent logs those two segments may not end up in same line.

````text
[TRACE]
I am log from other package
realis auroraResponse{ 100 pointer address}
````

 And for some longer, `Print` is just a shim for `Info` and will always create a new line regardless of `ln` i.e. [logrus](https://github.com/sirupsen/logrus/blob/master/logger.go#L201-L205)

Reply:
Yup, both you are right. There is a bit of reasoning behind why I did it this way though. 

For Printf functions it's pretty trivial to add a string in front of the message being printed out before it's flushed down to the logger but it does involve modifying the format string passed to prepend the level info.

For Print and Println functions however, it means I have to add an item to the array in the front of the incoming "array" (it's a variadic argument) or I can simply add it to the front of the first element. Adding in front of the first element will probably mean that an entire new array is allocated (I have yet to confirm this though!). 

Originally I wanted to stick to not modifying the arguments passed by the user but I don't really see too much harm in doing so.

Reply:
Since many logger are now leveled, most of the time log will end up like where the level and timestamp are always prepended by the underlying logger. There are duplication in level, though we could say the `TRACE` one is just the library's own verbose level and `INFO` is how the application developer things the level should be.

````
[INFO] 2018-01-10 TRACE: aurora.Response
````

For print it's a prepend, we are not modifying the arguments passed by user, the only draw back is you are creating some extra allocation because it's prepend instead of append (and even for append most cases I think we are creating new slice because the original one would have same length and cap for var args (I suppose ... didn't check the spec))

https://play.golang.org/p/hpdfX5QREYm

````go
package main

import (
	"log"
)

func myLog(args ...interface{}) {
	// log.Print("TRACE", args...)
	newArgs := []interface{}{"TRACE "}
	newArgs = append(newArgs, args...)
	log.Print(newArgs...)
}

func main() {
	myLog("has ", "a ", "prefix")
}
````